### ISSUE TITLE
ProtonMail support

### ISSUE THREAD

**Issue by dillonfagan** [elementary/mail#323]
It would be great if Mail on elementary OS were the first email client with built-in support for ProtonMail. In my opinion it's in the best interest of the Linux (and elementary) community to support Proton's efforts to protect individual privacy online. Building in support for and promoting the use of ProtonMail within the Mail app would be phenomenal.

**Comment by eljefuri** [comment]
Can you define precisely what you mean by support? Does ProtonMail use different protocols than IMAP/POP3/SMTP, or are you talking about the encryption aspect presumably covered by [this issue](https://github.com/elementary/mail/issues/82)? I have never looked ProtonMail myself so have no idea what their special sauce is, if anything.

**Comment by dillonfagan** [comment]
Yeah so ProtonMail does not use the standard email protocols, which is why with email clients today, you have to install a tool they developed called the ProtonMail Bridge. This grabs your emails with their secure protocol and then spoofs IMAP/SMTP on your local machine to then send those emails to your email client.

I am by no means an encryption or email expert, so I'm not entirely sure how Proton's security works. I do know that they use open source software and that emails are encrypted with AES. Maybe [this link](https://protonmail.com/security-details) can be helpful?

**Comment by eljefuri** [comment]
Given how Mail now works in master, I assume that this needs to be done as a [switchboard-plug-onlineaccounts](https://github.com/elementary/switchboard-plug-onlineaccounts/) integration. Mind you, I do not yet understand how *that* works, but I have a feeling that the issue does not belong here exactly.

**Comment by dillonfagan** [comment]
Oh, so the switchboard plugin actually does the fetching of emails?

**Comment by eljefuri** [comment]
I am honestly not sure yet. Just trying to understand all of this now.

**Comment by nblackburn** [comment]
https://protonmail.com/bridge

**Comment by DkGr** [comment]
Protonmail Bridge does not really support Mail. After configuring Mail with Protonmail Bridge, email are not well synchronized. Sometimes, folders are not synchronized or sometimes e-mail cannot be displayed (infinite loading). I don't know if these problems are caused by some missing features on Mail or if it's a wrong behaviour of Protonmail Bridge.

**Comment by nblackburn** [comment]
@DkGr Sorry to hear that, as far as i am aware this is the only means of supporting it as it uses it's own protocol which is proprietary.

**Comment by dillonfagan** [comment]
Maybe this could be helpful? https://github.com/ProtonMail/WebClient

**Comment by shawn1thompson** [comment]
They have been open-sourcing their apps. The Bridge is [here](https://github.com/ProtonMail/proton-bridge).

**Comment by marbetschar** [comment]
TL;DR For elementary to eventually support Proton Mail Synchronization, it needs to be added to Evolution Data Server first: https://gitlab.gnome.org/GNOME/evolution-data-server

Synchronization of PIM data (Mail, Calendar and Tasks) is done by the underlying Evolution Data Server. Online Accounts itself is just a fronted to Evolution Data Server for configuring the accounts it should use for synchronization.

**Comment by swt83** [comment]
Here to +1 the effort to get Protonmail working on Elementary, even if it was just using Bridge.

**Comment by victorodg** [comment]
I also wish I could use protonmail.. even if through bridge
